src/push/WMTPush.ts

```diff
@@ -16,30 +16,104 @@
 
 import { WMTNetworking, type WMTRequestProcessor, type WMTResponse } from "../networking/WMTNetworking"
 import { PowerAuthAuthentication } from 'react-native-powerauth-mobile-sdk'
-import { WMTPlatformUtils } from "../WMTPlatformUtils"
 
 /** Push handling */
 export class WMTPush extends WMTNetworking {
 
   /** 
-   * Registers the given powerauth activation for push notifications.
+   * Registers the PowerAuth activation for push notifications on the PowerAuth backend.
    * 
-   * @param token Push token.
-   * @param platform ios, android or huawei. When not specified, `Platform.OS` is used to determine if ios or android will be used. There is currently no automatic huawei platform detection.
+   * @param data Push platform and token retrieved from the device.
    * @param requestProcessor You may modify the request via this processor. It's highly recommended to only modify HTTP headers.
    * @returns Server response
+   * 
+   * ----
+   * For example, to register an FCM (Firebase Cloud Messaging) token, you can use:
+   * ```typescript
+   * const pushData = WMTPushPlatform.fcm("your_fcm_token")
+   * await mtoken.push.register(pushData)
+   * ```
+   * ----
+   *
+   * If you are using an older version of the Wultra Mobile Token API (1.9 or earlier), you may need to use the legacy format:
+   * ```typescript
+   * const pushData = WMTPushPlatform.fcm("your_fcm_token").supportLegacyServer()
+   * await mtoken.push.register(pushData)
+   * ```
    */
-  async register(token: string, platform?: "ios" | "android" | "huawei", requestProcessor?: WMTRequestProcessor): Promise<WMTResponse<void>> {
+  async register(data: WMTPushPlatform, requestProcessor?: WMTRequestProcessor): Promise<WMTResponse<void>> {
 
-    platform = platform || WMTPlatformUtils.getPlatform()
+    let anyData = data as any
 
     return await this.postSignedWithToken<void>(
-      { requestObject: { token: token, platform: platform } },
+      { requestObject: { token: anyData.token, platform: anyData.platform, environment: anyData.apnsEnvironment } },
```

Comment:
I was working way too long with typescript nerds to approve `as any` without at least a suggestion ;)
```suggestion
    data.requestObject,
```
This still prevents exposing the internals and removes the `any` dance.

Reply:
Yea, probably a good idea...